core/embed/rust/src/ui/model_mercury/component/result.rs

```diff
@@ -0,0 +1,166 @@
+        Self {
+            fg_color,
+            bg_color,
+            divider_color,
+        }
+    }
+
+    pub const fn message_style(&self) -> TextStyle {
+        TextStyle::new(Font::NORMAL, self.fg_color, self.bg_color, FG, FG)
+    }
+
+    pub const fn title_style(&self) -> TextStyle {
+        TextStyle::new(Font::BOLD, self.fg_color, self.bg_color, FG, FG)
+    }
+}
+
+pub struct ResultFooter<'a> {
+    style: &'a ResultStyle,
+    text: Label<'a>,
+    area: Rect,
+}
+
+impl<'a> ResultFooter<'a> {
+    pub fn new(text: Label<'a>, style: &'a ResultStyle) -> Self {
+        Self {
+            style,
+            text,
+            area: Rect::zero(),
+        }
+    }
+
+    pub const fn split_bounds() -> (Rect, Rect) {
+        let main_area = Rect::new(
+            Point::new(RESULT_PADDING, 0),
+            Point::new(WIDTH - RESULT_PADDING, RESULT_FOOTER_START),
+        );
+        let footer_area = Rect::new(
+            Point::new(RESULT_PADDING, RESULT_FOOTER_START),
+            Point::new(WIDTH - RESULT_PADDING, screen().height()),
+        );
+        (main_area, footer_area)
+    }
+}
+
+impl Component for ResultFooter<'_> {
+    type Msg = Never;
+
+    fn place(&mut self, bounds: Rect) -> Rect {
+        self.area = bounds;
+        self.text.place(bounds);
+        bounds
+    }
+
+    fn event(&mut self, _ctx: &mut EventCtx, _event: Event) -> Option<Self::Msg> {
+        None
+    }
+
+    fn paint(&mut self) {
+        // divider line
+        let bar = Rect::from_center_and_size(
+            Point::new(self.area.center().x, self.area.y0),
+            Offset::new(self.area.width(), 1),
+        );
+        display::rect_fill(bar, self.style.divider_color);
+
+        // footer text
+        self.text.paint();
+    }
+}
+
+pub struct ResultScreen<'a> {
+    bg: Pad,
+    footer_pad: Pad,
+    style: &'a ResultStyle,
+    icon: Icon,
+    message: Child<Label<'a>>,
+    footer: Child<ResultFooter<'a>>,
+}
+
+impl<'a> ResultScreen<'a> {
+    pub fn new(
+        style: &'a ResultStyle,
+        icon: Icon,
+        message: TString<'a>,
+        footer: Label<'a>,
+        complete_draw: bool,
+    ) -> Self {
+        let mut instance = Self {
+            bg: Pad::with_background(style.bg_color),
+            footer_pad: Pad::with_background(style.bg_color),
+            style,
+            icon,
+            message: Child::new(Label::centered(message, style.message_style())),
+            footer: Child::new(ResultFooter::new(footer, style)),
+        };
+
+        if complete_draw {
+            instance.bg.clear();
+        } else {
+            instance.footer_pad.clear();
+        }
+        instance
+    }
+}
+
+impl<'a> Component for ResultScreen<'a> {
+    type Msg = Never;
+
+    fn place(&mut self, _bounds: Rect) -> Rect {
+        self.bg.place(screen());
+
+        let (main_area, footer_area) = ResultFooter::<'a>::split_bounds();
```

Comment:
bump your version of `result.rs` and `error.rs` to get rid of these lifetimes

Reply:
9f4dc34cb76dfeb25f98d31258977bad3d35e112